observ/watcher.py

```diff
@@ -129,6 +129,8 @@ class Watcher:
         "_number_of_callback_args",
         "__weakref__",
     ]
+    on_created = None
```

Comment:
This would benefit from type annotations, so maybe wait for #111 ? Or otherwise we'll need to update it in that PR.

Reply:
Added type hints